### PR TITLE
5505: Adds global SMS key

### PR DESCRIPTION
**Pull request by meltheadorable**

### CODE REVIEW

File: app/controllers/settings_controller.rb [hunk 1]
```diff
@@ -35,7 +35,11 @@ def regenerate_incoming_sms_token
   end
 
   def using_incoming_sms_token_message
-    url = mission_sms_submission_url(@setting.incoming_sms_token, locale: nil)
+    if params[:missionless].present?
+      url = missionless_sms_submission_url(@setting.universal_sms_token, locale: nil, mission_name: nil, mode: nil)
+    else
+      url = mission_sms_submission_url(@setting.incoming_sms_token, locale: nil)
+    end
```
Comment (hooverlunch) [inline app/controllers/settings_controller.rb]: Ah, nice that you included this modal stuff!

File: app/controllers/sms_controller.rb [hunk 2]
```diff
@@ -40,7 +40,7 @@ def create
     end
 
     # Mission is guaranteed to be set by this point.
-    if params[:token] != current_mission.setting.incoming_sms_token
+    unless [current_mission.setting.incoming_sms_token, configatron.universal_sms_token].include? params[:token]
```
Comment (hooverlunch) [inline app/controllers/sms_controller.rb]: What if setting.incoming_sms_token is nil and params[:token] is nil?. Perhaps add a compact?

File: config/initializers/local_config.rb.example [hunk 3]
```diff
@@ -37,6 +37,7 @@ ActionMailer::Base.smtp_settings = {
 # ** FrontlineCloud **
 # configatron.default_settings.outgoing_sms_adapter = "FrontlineCloud"
 # configatron.default_settings.frontlinecloud_api_key = "xxx"
+# configatron.universal_sms_token = "50b488c71be4e9cd0ba34a7bf0b7053f"
```
Comment (hooverlunch) [inline config/initializers/local_config.rb.example]: This looks like it's part of the frontline cloud settings. Perhaps put in a separate paragraph and add a comment?

File: config/locales/en.yml [hunk 4]
```diff
@@ -546,6 +547,8 @@ en:
         incoming_sms_numbers_str: "Enter the phone number(s) (including country code, separated by commas) to which incoming SMSes for SMS forms should be sent. Only needed if using SMS submissions."
         incoming_sms_token: |-
           This token is included in the URL used by the incoming SMS provider to prevent the submission of unauthorized messages.
+        universal_sms_token: |-
+          This token is included in the URL used by the incoming SMS provider to prevent the submission of unauthorized messages. Use this URL if you need to allow multiple missions to submit via the same URL.
```
Comment (hooverlunch) [inline config/locales/en.yml]: Nice.

File: spec/support/incoming_sms_support.rb [hunk 5]
```diff
@@ -53,10 +53,11 @@ def do_incoming_request(params)
     req_env = {}
 
     url_prefix = defined?(missionless_url) && missionless_url ? "" : "/m/#{get_mission.compact_name}"
+    url_token = defined?(missionless_url) && missionless_url ? configatron.universal_sms_token : get_mission.setting.incoming_sms_token
 
     params[:sent_at] ||= Time.now
     params[:incoming][:adapter] ||= "TwilioSms"
-    params[:url] ||= "#{url_prefix}/sms/submit/#{get_mission.setting.incoming_sms_token}"
+    params[:url] ||= "#{url_prefix}/sms/submit/#{url_token}"
```
Comment (hooverlunch) [inline spec/support/incoming_sms_support.rb]: I think we should have a quick spec that checks that two different missions (i.e. forms from two different missions) can submit using the missionless URL. Not checking this was why I missed this issue in the first place. Could you add?

File: app/controllers/sms_controller.rb (hunk 2 above)
Comment (meltheadorable) [inline app/controllers/sms_controller.rb]: Good thinking.

File: spec/support/incoming_sms_support.rb (hunk 5 above)
Comment (meltheadorable) [inline spec/support/incoming_sms_support.rb]: Yep.

**Pull request merged**